### PR TITLE
Extract out SiteNestedContainer to its own file

### PR DESCRIPTION
**Pull request by hughbe**
No cleanup, only copy-and-paste. Line diffs are due to differences in the number of usings

Rationale is because the file is not only used in `DesignerHost` but also `DesignerSurface` so better belongs on its own file

### CODE REVIEW

**Comment by hughbe** [comment]
Hmmm 
image

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                return ownerName;
+            }
+        }
+
+        /// <summary>
+        /// Called to add a component to its container.
+        /// </summary>
+        public override void Add(IComponent component, string name)
+        {
+            if (_host.AddToContainerPreProcess(component, name, this))
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs]: Please invert to reduce nesting

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            }
+            return new NestedSite(component, _host, name, this);
+        }
+
+        /// <summary>
+        /// Called to remove a component from its container.
+        /// </summary>
+        public override void Remove(IComponent component)
+        {
+            if (_host.RemoveFromContainerPreProcess(component, this))
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs]: ditto

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+        protected override object GetService(Type serviceType)
+        {
+            object service = base.GetService(serviceType);
+            if (service != null)
+            {
+                return service;
+            }
+
+
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs]: [nit] unnecessary blank line

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+                }
+                return _services;
+            }
+
+
+            if (_services != null)
+            {
+                return _services.GetService(serviceType);
+            }
+            else
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs]: This `else` is redundant

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            {
+                if (Owner.Site != null && _safeToCallOwner)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs]: Please invert

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+            {
+                _container = container as SiteNestedContainer;
+                _name = name;
+            }
+
+            public string FullName
+            {
+                get
+                {
+                    if (_name != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/SiteNestedContainer.cs]: Please invert

**Comment by hughbe** [comment]
PR feedback addressed and have added tests :)

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerHost.cs [hunk 7]
```diff
@@ -1857,8 +1857,7 @@ string ISite.Name
 
                         if (validateName)
                         {
-                            INameCreationService nameService = (INameCreationService)((IServiceProvider)this).GetService(typeof(INameCreationService));
-                            if (nameService != null)
+                            if (((IServiceProvider)this).GetService(typeof(INameCreationService)) is INameCreationService nameService)
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerHost.cs]: #817

**Comment by hughbe** [comment]
@zsd4yr good to merge?

**Pull request merged**